py/misc.h

```diff
@@ -259,20 +259,12 @@ typedef union _mp_float_union_t {
 
 /** ROM string compression *************/
 
-#ifdef NO_QSTR
+#if MICROPY_ROM_TEXT_COMPRESSION && defined(NO_QSTR)
```

Comment:
I think this would be clearer to structure more like qstr.c (which has the same three cases).

```
#if MICROPY_ROM_TEXT_COMPRESSION

#ifdef NO_QSTR

// Compression, enabled but doing extraction.

#else

// Regular build.

#endif // NO_QSTR

#else

// Compression disabled.

#endif // MICROPY_ROM_TEXT_COMPRESSION
```

Reply:
The reason I did it the way it is, is to eliminate nesting of #if's, which can get rather confusing because there's no indentation here.

Reply:
Ok, this is now changed to match qstr.c.